girder_tech_journal/__init__.py

```diff
@@ -36,13 +36,52 @@
 from girder.models.user import User
 from girder.utility import mail_utils
 from girder.utility.model_importer import ModelImporter
+from girder.plugin import GirderPlugin
 from girder import events
 from girder_worker_utils.transforms.girder_io import GirderUploadToItem
 from tech_journal_tasks.tasks import processGithub, surveySubmission
 from . import constants
 from .models.journal import download_statistics, Journal
 
 
+class TechJournalPlugin(GirderPlugin):
+    DISPLAY_NAME = 'Tech Journal'
+    CLIENT_SOURCE_PATH = 'web_client'
+
+    def load(self, info):
+        techJournal = TechJournal()
+        techJournal.download_statistics = download_statistics()
+        techJournal.journal_collection = Journal()
+        ModelImporter.registerModel('journal', Journal, plugin='tech_journal')
+        mail_utils.addTemplateDirectory(os.path.join(
+            os.path.dirname(__file__),
+            'mail_templates'
+        ))
+
+        events.bind('model.setting.validate', 'journalMain', validateSettings)
+        info['apiRoot'].journal = techJournal
+        info['config']['/tech_journal'] = {
+            'tools.staticdir.on': True,
+            'tools.staticdir.dir': os.path.join(
+                os.path.abspath('girder-tech-journal-gui'), 'dist'),
```

Comment:
I'm not sure if this is the best way to go about locating the built dist directory, but it seems to always be running from the plugin root directory, so this worked for me.

Reply:
I think `pkg_resources.resource_filename` may help you here. See how we use it in Girder: https://github.com/girder/girder/blob/eb82b05608019cec6669e2c1b11b995191393b17/girder/cli/build.py#L21

Reply:
Then again, maybe it won't in the case that we're pip installing this. We might need to discuss with @jbeezley the best way to distribute the standalone front-end code.